src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -221,4 +222,16 @@ public static String parseDescription(String description) throws ParseException
         }
         return trimmedDescription;
     }
+
+    /**
+     * Parses a {@code String} into a {@code Goal.Frequency}
+     *
+     * @throws ParseException if the given {@code String} is an invalid frequency.
+     */
+    public static Goal.Frequency parseFrequency(String frequencyString) throws ParseException {
+        if (!Goal.ENUM_MAP.containsKey(frequencyString)) {
+            throw new ParseException("Invalid frequency supplied");
+        }
+        return Goal.ENUM_MAP.get(frequencyString);
```

Comment:
Same for this. `Goal.getFrequency(frequencyString)` would probably be better.

Reply:
Another way would be to convert into one method `Goal.parseFrequency(frequencyString)` that throws an exception if the `frequencyString` is invalid. I would prefer this as it would reduce the API exposed from `Goal`.